ecommerce/api.py

```diff
@@ -107,6 +109,37 @@ def generate_checkout_payload(request):
     return payload
 
 
+def check_discount_for_products(discount, basket):
+    """
+    Checks the validity of the discount against what's in the basket.
+
+    If the discount either has no products associated with it, or the products
+    in the basket are applicable to the discount, this returns True.
+    Otherwise, returns False.
+
+    Args:
+        - basket (Basket): the current basket
+        - discount (Discount|string: the discount to apply (if a string, loads the discount code specified)
+    Returns:
+        boolean
+    """
+    if not isinstance(discount, Discount):
+        discount = Discount.objects.filter(discount_code=discount).first()
+
+    basket_item_products = [item.product for item in basket.basket_items.all()]
```

Comment:
I think this could be put into a model method like `get_all_products`or ` get_basket_products`, since somewhere else in the code you are also getting all the products from a basket, like here https://github.com/mitodl/mitxonline/blob/main/ecommerce/models.py#L339. So

Reply:
Got this done in https://github.com/mitodl/mitxonline/pull/708/commits/0da926720553395f9c4c9a6506dcecc3b9596699